### DISCUSSION TITLE
How can I export & import in one line ?

### DISCUSSION THREAD

**Discussion by bowenxuuu** [slint-ui/slint#7164] [Q&A]
My code:
```
import { VerticalBox } from "std-widgets.slint";

import { Head } from "head.slint";
import { Body } from "body.slint";
import { Modal } from "modal.slint";
export { Template } from "template.slint";

import { Basket, Param, Stage, ModalData } from "../model/basket.slint";
export { Basket, Param, Stage, ModalData }

```

Is there a way to both `export` and `import` in one line of code? Just like this: 
`export import { Basket, Param, Stage, ModalData } from "../model/basket.slint";`
_or_
`import export { Basket, Param, Stage, ModalData } from "../model/basket.slint";`

**Comment by tronical** [comment]
You can write `export { Basket, Param, Stage, ModalData } from "../model/basket.slint";`

**Reply by bowenxuuu** [reply]
![1111](https://github.com/user-attachments/assets/442522b6-5ef6-462a-b650-cacea6751e38)


Error: Unknown type 'ModalData'

**Reply by tronical** [reply]
Yes, this merely exports. If you want to also use it locally, then you have to import it as well - in that case there's no way around doing both.

**Reply by ogoffart** [reply]
Should `export` be changed to also import it?

**Reply by bowenxuuu** [reply]
Could you please add support for 
`export import { SomeThing } from "some_module.slint";` ?


This is correctly in C++ 20 (Modules):
```
export module A;
export import SomeModule;
void some_fn(){}
```

In Rust, the keywords `use` and `pub use` are used to achieve similar effects.
